core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -585,6 +600,62 @@ extern "C" fn new_show_error(n_args: usize, args: *const Obj, kwargs: *mut Map)
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_webauthn(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let icon: Option<StrBuffer> = kwargs.get(Qstr::MP_QSTR_icon)?.try_into_option()?;
+        let app_names: Obj = kwargs.get(Qstr::MP_QSTR_app_names)?;
+        let account_name: Obj = kwargs.get(Qstr::MP_QSTR_account_names)?;
+
+        // Need to find the overall page count for pagination purposes.
+        let page_count = {
+            let mut count = 0;
+            let mut iter_buf = IterBuf::new();
+            let iter = Iter::try_from_obj_with_buf(app_names, &mut iter_buf)?;
+            for _ in iter {
+                count += 1;
+            }
+            count
+        };
+
+        // Closure to lazy-load the information on given page index.
+        // Done like this to allow arbitrarily many pages without
+        // the need of any allocation here in Rust.
+        let get_page = move |page_index| {
+            let app_name: StrBuffer = {
+                let mut iter_buf = IterBuf::new();
+                let mut iter = unwrap!(Iter::try_from_obj_with_buf(app_names, &mut iter_buf));
+                iter.nth(page_index).unwrap().try_into().unwrap()
+            };
+
+            let account_name: Option<StrBuffer> = {
+                let mut iter_buf = IterBuf::new();
+                let mut iter = unwrap!(Iter::try_from_obj_with_buf(account_name, &mut iter_buf));
+                iter.nth(page_index).unwrap().try_into_option().unwrap()
+            };
+
+            FidoPage::new(app_name, account_name)
+        };
+        let pages = FidoPages::new(get_page, page_count, "_tmp".into());
```

Comment:
Nit: you can make make `FidoPaginatedPage::new` accept `get_page` and `page_count` and turn `FidoPages` into private structure, without _tmp it doesn't serve any purpose to have it as part of API.

Reply:
Nice suggestion, in the end I got completely rid of `FidoPages`, as the `page_count` is stored in `scrollbar` and therefore `FidoPaginatedPage` can just store `get_page` function - [b557d76](https://github.com/trezor/trezor-firmware/pull/2556/commits/b557d76468ea90567cba981b4288e96502a5b73f)